### PR TITLE
feat(forms): add option to use native validation and angular forms

### LINKED ISSUE #13573: Add option to use browser's native validation and Angular forms
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
As described here https://github.com/angular/angular/pull/13092#issuecomment-267900308 currently It's not possible to enable native validate behavior, while still having Angular forms functionality.

**Expected behavior**
It should be possible to use browser's native validation and Angular forms.

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes https://github.com/angular/angular/issues/13573

### CODE REVIEW

File: modules/@angular/forms/src/directives/ng_novalidate_directive.ts [hunk 1]
```diff
@@ -8,8 +8,19 @@
...
+ *
+ * If you want to use native validation with angular forms just add `ngNoNovalidate` attribute:
+ *
+ * ```
+ * <form ngNoNovalidate></form>
+ * ```
+ */
 @Directive({
-  selector: 'form:not([ngNoForm])',
+  selector: 'form:not([ngNoForm]):not([ngNoNovalidate])',
```
Comment (vicb) [inline modules/@angular/forms/src/directives/ng_novalidate_directive.ts]: I would use `ngNoNoValidate` ?

Comment (vicb) [inline modules/@angular/forms/src/directives/ng_novalidate_directive.ts]: ... or `ngNativeValidate` ?

**Comment by vicb** [comment]
@DzmitryShylovich when adding new features, could you please make sure to create a GH issue on angular/angular.io to ask for it to be documented on top of adding API docs to the source code ?

Thanks

File: modules/@angular/forms/src/directives/ng_novalidate_directive.ts (hunk 1 above)
Comment (DzmitryShylovich) [inline modules/@angular/forms/src/directives/ng_novalidate_directive.ts]: I chose `ngNoNoValidate` to be consistent with `ngNoForm` and others. But I really like your proposal.

**Comment by vicb** [comment]
Could you please also rename `NgNovalidate` to `NgNoValidate` (thanks @chuckjaz for catching this)

**Comment by DzmitryShylovich** [comment]
@vicb done

**Comment by vicb** [comment]
@DzmitryShylovich could you please mention the rename in the commit comment ? Have you opened an issue on angular/angular.io ?

**Comment by DzmitryShylovich** [comment]
> Have you opened an issue on angular/angular.io ?

https://github.com/angular/angular/issues/13573

**Comment by DzmitryShylovich** [comment]
> could you please mention the rename in the commit comment ?

done

**Comment by vicb** [comment]
The directive should be exported and the .d.ts updated

**Comment by DzmitryShylovich** [comment]
@vicb why? It's a private api.

**Comment by DzmitryShylovich** [comment]
@vicb I exported it

**Comment by mhevery** [comment]
@kara can you approve this?

File: modules/@angular/forms/src/directives/ng_no_validate_directive.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+import {Directive} from '@angular/core';
+
+/**
+ * @whatItDoes Adds `novalidate` attribute to all forms by default.
+ *
+ * `novalidate` is used to disable browser's native form validation.
+ *
+ * If you want to use native validation with angular forms just add `ngNativeValidate` attribute:
```
Comment (kara) [inline modules/@angular/forms/src/directives/ng_no_validate_directive.ts]: Super nit: Can you add a comma? :)

```If you want to use native validation with Angular forms, just add `ngNativeValidate` attribute:```

**Review by kara — APPROVED** [review]
LGTM. Added a nit in case you want to fix.

**Comment by DzmitryShylovich** [comment]
@kara done

**Pull request merged**